app/src/main/java/com/example/campusguide/directions/Route.kt

```diff
@@ -19,18 +23,58 @@ import kotlinx.coroutines.launch
  * @param end The name of the location where the route ends
  */
 class Route constructor(private val map: GoogleMap, private val activity: AppCompatActivity) {
+    private val PATTERN_DASH_LENGTH_PX = 20
+    private val PATTERN_GAP_LENGTH_PX = 20
+    private val DASH: PatternItem = Dash(PATTERN_DASH_LENGTH_PX.toFloat())
+    private val GAP: PatternItem = Gap(PATTERN_GAP_LENGTH_PX.toFloat())
+    private val PATTERN_POLYGON_ALPHA = listOf(GAP, DASH)
+    private val COLOR_BLUE_ARGB = 0xff0000ff
     private var polyline: Polyline? = null
+    private var begin: Marker? = null
+    private var dest: Marker? = null
+
+    private fun capitalizeWords(location: String): String {
+        val words = location.split(" ").toMutableList()
+        var output = ""
+        for (word in words) {
+            output += word.capitalize() + " "
+        }
+        output = output.trim()
+        return output
+    }
 
     fun set(start: String, end: String) {
         polyline?.remove()
-
+        begin?.remove()
+        dest?.remove()
         val directions = Directions(activity)
 
         //Create a coroutine so we can invoke the suspend function Directions::getDirections
         GlobalScope.launch {
             val response = directions.getDirections(start, end)
-
-            if(response != null) {
+            if (response != null) {
+                val startPoint = MarkerOptions().position(
+                    LatLng(
+                        response.routes[0].legs[0].startLocation.lat.toDouble(),
+                        response.routes[0].legs[0].startLocation.lng.toDouble()
+                    )
+                ).title(capitalizeWords(start)).snippet("Start")
```

Comment:
"Start" and "Destination" should be in constant variables.

Reply:
Added this change request